system/HTTP/Response.php

```diff
@@ -257,6 +270,10 @@ public function __construct($config)
 		$this->cookieHTTPOnly = $config->cookieHTTPOnly;
 		$this->cookieSameSite = $config->cookieSameSite ?? $this->cookieSameSite;
 
+    // CORS
+    $this->CORSEnabled    = $config->CORSEnabled ?? $config->CORSEnabled;
+    $this->CORS 					= new CORS(new \Config\CORS());
```

Comment:
It is bc break, no `\Config\CORS()` yet in current 4.0.4 version.

Reply:
OK I will demonstrate with a scenario what will happen if this new config file is not handled properly.

Let us say I have a project using `v4.0.4` as the framework version. In this version there are no `CORS` class in `system/` yet and `Config\CORS` in `app/`. Let us assume further that I am using Composer so the files are in the `vendor/` directory. Now let us say the framework releases the new version `v4.0.5` this November. I got excited and updated my project to use the new version on that date. On my update, in the `vendor/` directory the `CORS` class from this PR is now found in `vendor/codeigniter4/framework/system/HTTP/`. The `Config\CORS` file also from this PR is now found in `vendor/codeigniter4/framework/app/Config/`. OK, good then. **BUT** take note that your `CORS` class **will not** look for the config file in `vendor/codeigniter4/framework/app/Config/`. It will **instead** look for the file in **my project's `app/Config/` directory**. But is that `Config\CORS` file in my project's `app/Config/` folder? Definitely no. So in the instantiation of the `CORS` class it will throw an error for class not found, effectively breaking my project.

So instead of being happy on upgrading to v4.0.5 for the new features I instead got a broken project. A completely failing project which I have no recourse but to revert to v4.0.4 just for my project to be alive.

I hope now you get what @samsonasik is trying to tell that this is a BC break without proper handling.

Reply:
Let me understand what you said. If someone updates a project with this new file `Config/CORS.PHP`the composer will put this file at `vendor` folder, not in `App/Config` of the project? OK, I agree, but how can this be added new config feature without creating an external library for this? Cause the intention here is adding this as a solution to the CORS problem when creating RESTful APIs and this need a configuration params as part of this solution, not as an external library but a core solution.

Reply:
Here you need to come up with a fallback solution to ensure compatibility with earlier versions. So instead of creating a new `Config\CORS` class you can put those settings as a separate section to `Config\App` class. Then in your main `CORS` class, you will have **default values** provided for the equivalent settings in config. So this is basically like the implementation in samesite: `$this->cookieSamesite = $config->cookieSamesite ?? $this->cookieSamesite;`.

No need for you to create an external lib for that. What I meant in my comment below was @agungsugiarto 's library will not cause BC breaks because it is external. His config is consumed by his library's main class which in turn is passed down to the core.

In your case, the config will be integrated to the core. So without proper precautions and handling, the disadvantages may outweigh the benefits so there's a need to have fallbacks. For earlier CI versions, these fallbacks will be their defense against BC before they can actually update the affected config files. Of course, this is not a problem with newer releases.

Reply:
I made some tests here and it's really will cause a BC BREAK cause `composer` will put this file in the `vendor` folder.

So, the `namespace` is correctly set, the `Config/CORS.php` it's really necessary to this solution of the core framework CORS error. 

So it can be rejected, as you want, or try to add this as a new version that will need a manual upgrade as is the versions 3 for 4.


For now, I'll up this solution to the CORS problem in my GitHub as a library that will need to manually and try to add this to the `composer`